include/types/serviceAttributes.class.php

```diff
@@ -29,19 +29,15 @@
 class serviceAttributes extends AbstractType {
 
     /**
-     * @var (object)serviceProvider
+     * @var optional (object)serviceProvider
      */
     public $serviceProvider;
 
     /**
-     * @var (object)service
+     * @var optional (object)service
      */
     public $service;
 
-    /**
```

Comment:
did not explicitly tell you to delete it but can you also delete the supprtedContentSelectionMethods class and remove it in classmap.php. it's dead code so not needed anymore.